### PR TITLE
Initial test cases for Canned Role functionality

### PR DESCRIPTION
**Pull request by oshtaier**
```
nosetests tests/foreman/ui/test_role.py
.........
----------------------------------------------------------------------
Ran 9 tests in 480.185s

OK
```

### CODE REVIEW

**Comment by omaciel** [comment]
@jyejare can you review these and provide comments to @oshtaier ?

File: tests/foreman/ui/test_role.py [hunk 1]
```diff
@@ -102,39 +102,26 @@ def test_positive_update_permission(self):
...
-
 
 class CannedRoleTestCases(UITestCase):
     """Implements Canned Roles tests from UI"""
 
-    @stubbed
+    @classmethod
+    def setUpClass(cls):
+        """Create Organization and Location to be used in tests
+        """
```
Comment (abalakh) [inline tests/foreman/ui/test_role.py]: nitpick: this docstring will fit in 1 line

**Comment by sghai** [comment]
ACK..

Leaving for @jyejare to review this quickly.

File: robottelo/ui/role.py [hunk 2]
```diff
@@ -38,28 +41,46 @@ def delete(self, name, really=True):
...
+        self.search(role_name)
+        self.click(locators['roles.dropdown'] % role_name)
+        self.click(locators['roles.add_permission'])
+        if resource_type:
+            self.assign_value(
+                locators['roles.select_resource_type'], resource_type)
+        if permission_list:
+            self.configure_entity(
+                permission_list, FILTER['filter_permission'])
+        # Verify whether 'Override' checkbox is in expected state. Putting it
```
Comment (jyejare) [inline robottelo/ui/role.py]: expected state ? it can be explained in more specific words.

File: robottelo/ui/role.py [hunk 3]
```diff
@@ -38,28 +41,46 @@ def delete(self, name, really=True):
...
+        # here as we can do it only during create procedure, not after
+        if override_check is True:
```
Comment (jyejare) [inline robottelo/ui/role.py]: Can you bit illustrate why this is required and in what cases ?

File: tests/foreman/ui/test_role.py [hunk 4]
```diff
@@ -199,12 +216,46 @@ def test_positive_create_non_overridable_filter(self):
...
+            )
+        with Session(self.browser, username, password) as session:
+            set_context(session, org=self.role_org)
+            set_context(session, loc=self.role_loc)
+            self.assertIsNone(session.nav.wait_until_element(
+                menu_locators['menu.content'], timeout=3))
+            self.assertIsNone(session.nav.wait_until_element(
+                menu_locators['menu.infrastructure'], timeout=3))
+            # check that we can access edit functionality at all
+            self.architecture.update(old_name='x86_64', new_name='x86_64')
```
Comment (jyejare) [inline tests/foreman/ui/test_role.py]: In this Testcase you would have check 
1. if the locations and organizations tab is not present in Architecture filter.
2. If the override checkbox is not present in the Architecture filter.

**Comment by jyejare** [comment]
ACK pending questions and comments.

File: robottelo/ui/role.py (hunk 3 above)
Comment (oshtaier) [inline robottelo/ui/role.py]: As stated in the next comment, for non-overridable filters, 'Override' checkbox should be absent on the page. In all other cases it should be present and be in unchecked state

File: tests/foreman/ui/test_role.py (hunk 4 above)
Comment (oshtaier) [inline tests/foreman/ui/test_role.py]: 1) Nope, I should not as there are no information about that in stab
2) Is verified by `override_check=False,`

**Comment by oshtaier** [comment]
Addressed all comments

File: tests/foreman/ui/test_role.py (hunk 4 above)
Comment (jyejare) [inline tests/foreman/ui/test_role.py]: The 1st assert statement says that the 'filter is created without taxonomies' and we can verify that from UI itself by simply checking if the tabs for location and organization are available or not.
Anyhow its not a blocker comment as you covered this from other end.

**Pull request merged**